lib/src/Uncertainty/Algorithm/MetaModel/FunctionalChaos/FunctionalChaosSobolIndices.cxx

```diff
@@ -294,63 +294,168 @@ Scalar FunctionalChaosSobolIndices::getSobolTotalIndex(const UnsignedInteger var
+      // Compute numerator
+      mustInclude = true;
+      if (multiIndices.normInfinite() == 0)
       {
-        Indices multiIndices(enumerateFunction(coefficientIndices[i]));
-        // Take into account only nonzero multi indices
-        if (*std::max_element(multiIndices.begin(), multiIndices.end()) > 0)
+        // Exclude the zero indices
+        mustInclude = false;
+      }
+      else
+      {
+        for (UnsignedInteger variableIndex = 0; variableIndex < inputDimension; ++variableIndex)
         {
-          totalVariance += coefficientI * coefficientI;
-          // Set the exponents corresponding to the group to zero
-          for (UnsignedInteger j = 0; j < orderSobolIndice; ++j)
-            multiIndices[variableIndices[j]] = 0;
-          // Now check that all the indices are zero
-          if (*std::max_element(multiIndices.begin(), multiIndices.end()) == 0)
-            covarianceVariables += coefficientI * coefficientI;
-        } // *std::max_element(multiIndices.begin(), multiIndices.end()) > 0
-      } // if coefficientI <> 0
+          // Check that each variable having a nonzero degree is in the group
+          if (multiIndices[variableIndex] > 0 and not variablesGroup.contains(variableIndex)) 
+          {
+            mustInclude = false;
+            break;
+          }
+        }
+      }
+      if (mustInclude) covarianceVariables += coefficientI * coefficientI;
     } // coefficientIndices[i] > 0
   } // Loop over the coefficients
   if (totalVariance > 0.0) return covarianceVariables / totalVariance;
   else return 0.0;
 }
 
 /* Sobol grouped total index accessor */
-Scalar FunctionalChaosSobolIndices::getSobolGroupedTotalIndex(const Indices & variableIndices,
+Scalar FunctionalChaosSobolIndices::getSobolGroupedTotalIndex(const Indices & variablesGroup,
+    const UnsignedInteger marginalIndex) const
+{
+  const UnsignedInteger inputDimension = functionalChaosResult_.getDistribution().getDimension();
+  if (!variablesGroup.check(inputDimension)) throw InvalidArgumentException(HERE) << "The variable indices of a Sobol indice must be in the range [0, dim-1] and must be different.";
+  const Sample coefficients(functionalChaosResult_.getCoefficients().getMarginal(marginalIndex));
+  const Indices coefficientIndices(functionalChaosResult_.getIndices());
+  const UnsignedInteger size = coefficients.getSize();
+  Scalar covarianceVariables = 0.0;
+  const EnumerateFunction enumerateFunction(functionalChaosResult_.getOrthogonalBasis().getEnumerateFunction());
+  // Sum the contributions of all the coefficients associated to a basis vector involving only the needed variables
+  Scalar totalVariance = 0.0;
+  const UnsignedInteger groupDimension = variablesGroup.getSize();
+  bool mustInclude;
+  for (UnsignedInteger i = 0; i < size; ++i)
+  {
+    if (coefficientIndices[i] > 0)
+    {
+      const Scalar coefficientI = coefficients(i, 0);
+      Indices multiIndices(enumerateFunction(coefficientIndices[i]));
```

Comment:
const

Reply:
Ok. I fixed the similar const issue at e008e7f54f23bedd45db149731931b48c32e5608.